src/lib.rs

```diff
@@ -333,7 +360,16 @@ pub struct Evaluator<'tcx> {
 
     /// The random number generator to use if Miri
     /// is running in non-deterministic mode
-    pub(crate) rng: Option<StdRng>
+    pub(crate) rng: Option<StdRng>,
+
+    /// Extra information needed for unwinding
+    /// This is 'None' when we're running in abort mode
+    pub(crate) cached_data: Option<CachedTypes<'tcx>>
```

Comment:
The name is way more generic than just unwinding... maybe call this `unwinding_data`?

Reply:
Also please add a trailing `,`

Reply:
My intention was to have this be used for any future types we might want to cache.

Reply:
Is creating types really that expensive? Do you add creating types on any hot paths that makes it worth it?
Given that panics are usually the cold path, I am not even sure if this is worth it here. Seems like maybe a premature optimization?

Reply:
It would be nice to be able to run [this test](https://github.com/rust-lang/rust/blob/efe2f32a6b8217425f361ec7c206910c611c03ee/src/liballoc/tests/slice.rs#L1602) under miri, which involves a large number of unwinds.

Reply:
Have you measured that the cache makes any difference there?

Also, that test uses `thread::spawn` -- though I guess it can be refactored to use `catch_unwind` instead.